File: 121_best_time_to_buy_and_sell_stock/121_best_time_to_buy_and_sell_stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int lowestPrice = Integer.MAX_VALUE;
        int todayProfit = 0;
        int maxProfit = 0;
        for (int i = 0; i < prices.length; i++){
            if(prices[i] < lowestPrice){
                lowestPrice = prices[i];
            }
            
            todayProfit = prices[i] - lowestPrice;
            
            if (todayProfit > maxProfit){
                maxProfit = todayProfit;
            }
        }
        return maxProfit;
    }
}

File: 121_best_time_to_buy_and_sell_stock/121_best_time_to_buy_and_sell_stock.py
import sys
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowest_price: int = sys.maxsize
        max_profit: int = 0
        today_profit: int = 0

        for i in range(len(prices)):
            if prices[i] < lowest_price:
                lowest_price = prices[i]

            today_profit = prices[i] - lowest_price

            if today_profit > max_profit:
                max_profit = today_profit

        return max_profit
